src/components/FilterList/FilterList.tsx

```diff
@@ -1,19 +1,56 @@
 import { Stack } from '@mantine/core';
 
+import { useWindowSize } from '@/windowmanagement/Window/hooks';
+
 import { FilterListData } from './FilterListData';
 import { FilterListFavorites } from './FilterListFavorites';
 import { FilterListInputField } from './FilterListInputField';
 import { FilterListProvider } from './FilterListProvider';
 
-interface FilterListProps {
-  children: React.ReactNode;
+interface BaseProps {
+  heightFunc?: (height: number) => number;
+  heightPercent?: number;
   height?: number | string;
+  children: React.ReactNode;
+  isLoading?: boolean;
+}
+
+interface FuncProps {
+  heightPercent?: never;
+  height?: never;
+}
+interface AbsoluteProps {
+  heightPercent?: never;
+  heightFunc?: never;
 }
 
-export function FilterList({ height = '100%', children }: FilterListProps) {
+interface RelativeProps {
+  height?: never;
+  heightFunc?: never;
+}
+
+type Props =
+  | (BaseProps & FuncProps)
+  | (BaseProps & AbsoluteProps)
+  | (BaseProps & RelativeProps);
+
+export function FilterList({
+  height,
+  heightFunc,
+  heightPercent,
+  isLoading,
+  children
+}: Props) {
+  const { height: windowHeight } = useWindowSize();
+
+  const calculatedHeight =
+    height ??
+    (heightPercent && windowHeight * (heightPercent / 100.0)) ??
+    (heightFunc && heightFunc(windowHeight));
```

Comment:
This is a minor one but I think the linter will complain about this syntax, I believe the correct one is to do `heightFunc?.(windowHeight)`, in any case I like what you did with the props here 👍